src/avatar.cpp

```diff
@@ -1461,6 +1461,19 @@ bool avatar::invoke_item( item *used, const std::string &method )
     return Character::invoke_item( used, method );
 }
 
+void avatar::update_cardio_acc()
+{
+    const int prev_cardio_acc = get_cardio_acc();
+    const int last_24h_kcal = calorie_diary.begin()->spent;
```

Comment:
```suggestion
    const int last_24h_kcal = calorie_diary.front().spent;
```

Since you don't need an iterator here, `front()` would be more appropriate I think.

Reply:
Also, if I'm not mistaken, the front of this array is only the spent calories since midnight - not the previous 24 hours.

Reply:
I don't think so:
```c++
    if( is_avatar() && ticks_between( from, to, 24_hours ) > 0 ) {
        as_avatar()->advance_daily_calories();
    }
```
in character_body.cpp and:
```c++
void avatar::advance_daily_calories()
{
    calorie_diary.push_front( daily_calories{} );
    if( calorie_diary.size() > 30 ) {
        calorie_diary.pop_back();
    }
}
```
in avatar.cpp.

Reply:
That's what I mean - every 24 hours, the front of this list gets a new (empty) calorie log. In order for this `update_cardio_acc` to work properly, I think it would need to be invoked from `Character::update_body` just *before* `advance_daily_calories()`, to capture the previous day's cardio level.